arduino-ide-extension/src/node/ide-updater/ide-updater-service-impl.ts

```diff
@@ -0,0 +1,131 @@
+import { injectable } from '@theia/core/shared/inversify';
+import { GenericServerOptions } from 'builder-util-runtime';
+import {
+  AppUpdater,
+  AppImageUpdater,
+  MacUpdater,
+  NsisUpdater,
+  UpdateInfo,
+  CancellationToken,
+} from 'electron-updater';
+import {
+  IDEUpdaterService,
+  IDEUpdaterServiceClient,
+} from '../../common/protocol/ide-updater-service';
+import fetch, { Response } from 'node-fetch';
+
+const CHANGELOG_BASE_URL = 'https://downloads.arduino.cc/arduino-ide/changelog';
+
+@injectable()
+export class IDEUpdaterServiceImpl implements IDEUpdaterService {
+  private updater: AppUpdater;
+  private cancellationToken?: CancellationToken;
+  protected theiaFEClient?: IDEUpdaterServiceClient;
+
+  constructor() {
+    const options: GenericServerOptions = {
+      provider: 'generic',
+      url: 'https://downloads.arduino.cc/arduino-ide/nightly/test/',
```

Comment:
Is this URL correct? If so, can we align the folder name with the channel name?

Reply:
We're using that one for testing purposes for now, it will be changed before merge.